addons/im_livechat/static/tests/embed/chat_window.test.js

```diff
@@ -114,3 +114,27 @@ test("avatar url contains access token for non-internal users", async () => {
         }/avatar_128?access_token=${guest.id}&unique=${deserializeDateTime(guest.write_date).ts}"]`
     );
 });
+
+test("Open confirm dialog on Esc, cancel it on Esc/ confirm it on Enter", async () => {
+    await startServer();
+    await loadDefaultEmbedConfig();
+    onRpcBefore((route, args) => {
+        if (route === "/im_livechat/visitor_leave_session") {
+            step(route);
+        }
+    });
+    await mountWithCleanup(LivechatButton);
+    await click(".o-livechat-LivechatButton");
+    await contains(".o-mail-ChatWindow");
+    await insertText(".o-mail-Composer-input", "Hello");
+    await triggerHotkey("Enter");
+    await contains(".o-mail-Message", { text: "Hello" });
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation");
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation", { count: 0 });
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation");
+    await triggerHotkey("Enter");
```

Comment:
Assert on just classname is not great. For core components like `Message` and `Thread` then ok, but `CloseConfirmation` is very niche component.

I would prefer to have at the very least the text, which is the most relevant part that the end-user also see, i.e. `"Leaving will end the livechat. Proceed leaving?"`.